tests/testArgumentParser.py

```diff
@@ -298,46 +297,35 @@ def testAtFile(self):
         self.assertEqual(namespace.config.floatItem, 4.7)
         self.assertEqual(namespace.config.strItem, "new value")
 
-    def testLogDest(self):
-        """Test --logdest
-        """
-        logFile = tempfile.NamedTemporaryFile()
-        try:
-            namespace = self.ap.parse_args(
-                config=self.config,
-                args=[DataPath, "--logdest", logFile.name],
-            )
-            namespace.log.info('test logging to the specified file')
-            self.assertTrue(os.path.isfile(logFile.name))
-            self.assertGreater(os.path.getsize(logFile.name), 0)
-        finally:
-            logFile.close()
-
     def testLogLevel(self):
         """Test --loglevel"""
-        for logLevel in ("debug", "Info", "WARN", "fatal"):
-            intLevel = getattr(pexLog.Log, logLevel.upper())
+        for logLevel in ("trace", "debug", "Info", "WARN", "eRRoR", "fatal"):
+            intLevel = getattr(lsstLog.Log, logLevel.upper())
             print("testing logLevel=%r" % (logLevel,))
             namespace = self.ap.parse_args(
                 config=self.config,
                 args=[DataPath, "--loglevel", logLevel],
             )
-            self.assertEqual(namespace.log.getThreshold(), intLevel)
+            self.assertEqual(namespace.log.getLevel(), intLevel)
             self.assertFalse(hasattr(namespace, "loglevel"))
 
-            fooBarLevel = intLevel + 27
-            bazLevel = -500
+            bazLevel = "TRACE"
             namespace = self.ap.parse_args(
                 config=self.config,
-                args=[DataPath, "--loglevel", str(intLevel),
-                      "foo.bar=%d" % (fooBarLevel,),
-                      "baz=0",
+                args=[DataPath, "--loglevel", logLevel,
+                      "foo.bar=%s" % (logLevel,),
+                      "baz=INFO",
                       "baz=%s" % bazLevel,  # test that later values override earlier values
                       ],
             )
-            self.assertEqual(namespace.log.getThreshold(), intLevel)
-            self.assertEqual(namespace.log.getThresholdFor("foo.bar"), fooBarLevel)
-            self.assertEqual(namespace.log.getThresholdFor("baz"), bazLevel)
+            self.assertEqual(namespace.log.getLevel(), intLevel)
+            self.assertEqual(lsstLog.Log.getLogger("foo.bar").getLevel(), intLevel)
+            self.assertEqual(lsstLog.Log.getLogger("baz").getLevel(), getattr(lsstLog.Log, bazLevel))
+
+        self.assertRaises(SystemExit, self.ap.parse_args,
+                          config=self.config,
+                          args=[DataPath, "--loglevel", "1234"],
+                          )
```

Comment:
Please use `with self.asserRaises` (and in other places in this file). It makes it much clearer what is being tested.